File: packages/sdk-android/android/traceitx-core/src/main/kotlin/com/traceitx/config/ConfigValidator.kt
// SPDX-License-Identifier: MIT
// SPDX-FileCopyrightText: 2026 ScriptX
//
// Validates a TraceItXConfig before TraceItX.start(...) installs it. Throws
// TraceItXConfigError on bad input — synchronously, so misconfiguration fails
// at app boot (not at runtime).
//
// Endpoint scheme validation is gone: the ingest URL is no longer a public
// config field. It's baked into BuildConfig.INGEST_URL per-variant (release
// = prod, debug = TRACEITX_DEV_INGEST_URL env). See IngestEndpoint.kt.
package com.traceitx.config

object ConfigValidator {
    fun validate(config: TraceItXConfig) {
        if (config.appId.isBlank()) throw TraceItXConfigError.MissingAppId
        if (config.sdkKey.isBlank()) throw TraceItXConfigError.BlankSdkKey
        val rate = config.vitals.sampleRate
        if (rate != null && (rate.isNaN() || rate !in 0.0..1.0)) throw TraceItXConfigError.InvalidVitalsSampleRate
    }
}

File: packages/sdk-ios/Sources/TraceItX/Config/ConfigValidator.swift
// SPDX-License-Identifier: MIT
// SPDX-FileCopyrightText: 2026 ScriptX
//
// Validates a TraceItXConfig before TraceItX.shared.start(_:) installs it.
// Throws TraceItXConfigError on bad input.
//
// Endpoint scheme validation is gone: the ingest URL is no longer a public
// config field. The URL is baked at compile time via IngestEndpoint.swift
// (Release builds get https://traceitx.com; Debug builds may read the
// TRACEITX_DEV_INGEST_URL env var).
import Foundation

enum ConfigValidator {
    static func validate(_ config: TraceItXConfig) throws {
        // Per Phase 04.2 D-03: hard-fail at start() if the SDK key is missing,
        // wrong-prefix, or wrong-length. Reuses .missingAppId verbatim — the
        // appId/sdkKey rename is explicitly deferred (D-07). Length 41 = prefix
        // "txx_live_" (9) + 32-char body matching admin SdkKeysPanel emission.
        let trimmed = config.appId.trimmingCharacters(in: .whitespacesAndNewlines)
        if trimmed.isEmpty || !trimmed.hasPrefix("txx_live_") || trimmed.count != 41 {
            throw TraceItXConfigError.missingAppId
        }
        // Session Vitals (iOS spec 2026-09-05 §1) — a local sample-rate
        // override outside [0,1] (including NaN) is a caller bug, not a
        // silent clamp.
        if let rate = config.vitals.sampleRate, rate.isNaN || rate < 0 || rate > 1 {
            throw TraceItXConfigError.invalidVitalsSampleRate
        }
    }
}
